dbms/src/Storages/KVStore/MultiRaft/RegionMeta.cpp

```diff
@@ -495,6 +495,37 @@ RegionMeta::RegionMeta(metapb::Peer peer_, metapb::Region region, raft_serverpb:
     region_state.setRegion(std::move(region));
 }
 
+
+static const metapb::Peer & findPeer(const metapb::Region & region, UInt64 peer_id)
+{
+    for (const auto & peer : region.peers())
+    {
+        if (peer.id() == peer_id)
+        {
+            return peer;
+        }
+    }
+
+    throw Exception(
+        ErrorCodes::LOGICAL_ERROR,
+        "{}: peer not found in region, peer_id={} region_id={}",
+        __PRETTY_FUNCTION__,
+        peer_id,
+        region.id());
+}
+
+RegionMeta RegionMeta::genFromPb(metapb::Region && region, UInt64 peer_id, UInt64 index, UInt64 term)
```

Comment:
Could just call genFromMetaRegion, it has been named this way. See `cloneMetaRegion`

Reply:
Picked